tensilelite/Tensile/CustomKernels/Custom_Cijk_Alik_Bljk_HHS_BH_UserArgs_MT128x16x128_MI16x16x1_SN_K1_MIWT2_1_triple_buffer.s

```diff
@@ -0,0 +1,3569 @@
+v_mfma_f32_16x16x16_f16 acc[4:7], v[vgprValuB_X0_I0+0+2+0:vgprValuB_X0_I0+0+2+0+1], v[vgprValuA_X0_I0+4+2+0:vgprValuA_X0_I0+4+2+0+1], acc[4:7] // left value = acc[4+0:7+0]
+/* numPrefetchIter=0 */
+/* dataAtIterA=-2 numReadsIterA=2 skipReadsIterA=3 readsPerIterA=2 */
+/* dataAtIterB=-2 numReadsIterB=2 skipReadsIterB=3 readsPerIterB=1 */
+
+/* iter 2 */
+/*  grEndMfmaIndex:9, lwStartMfmaIndex:9, lwEndMfmaIndex:9  */
+/*  numMfmaForLR:1, syncPlrMfmaIndex:12  */
+/*  mfmaIndex:4  */
+/* localReadsVacancy: latencyLeft 2 */
+s_add_u32 s[sgprSrdB+0], s[sgprSrdB+0], s74        // gra SRD += inc(lower)
+s_waitcnt lgkmcnt(6)                               // wait for prior local read local write old=0, new=6 newLW=0 newLR=6
+v_mfma_f32_16x16x16_f16 acc[0:3], v[vgprValuB_X2_I0+0+0+0:vgprValuB_X2_I0+0+0+0+1], v[vgprValuA_X2_I0+0+0+0:vgprValuA_X2_I0+0+0+0+1], acc[0:3] // left value = acc[0+0:3+0]
+/*  mfmaIndex:5  */
+/* localReadsVacancy: latencyLeft 2 */
+s_addc_u32 s[sgprSrdB+1], s[sgprSrdB+1], s75       // gra SRD += inc(upper)
+v_mfma_f32_16x16x16_f16 acc[4:7], v[vgprValuB_X2_I0+0+0+0:vgprValuB_X2_I0+0+0+0+1], v[vgprValuA_X2_I0+4+0+0:vgprValuA_X2_I0+4+0+0+1], acc[4:7] // left value = acc[4+0:7+0]
+/* numPrefetchIter=0 */
+/* dataAtIterA=-1 numReadsIterA=2 skipReadsIterA=2 readsPerIterA=2 */
+/* dataAtIterB=-1 numReadsIterB=2 skipReadsIterB=2 readsPerIterB=1 */
+
+/* iter 3 */
+/*  grEndMfmaIndex:9, lwStartMfmaIndex:9, lwEndMfmaIndex:9  */
+/*  numMfmaForLR:1, syncPlrMfmaIndex:12  */
+/*  mfmaIndex:6  */
+/* localReadsVacancy: latencyLeft 2 */
+s_sub_u32 s[sgprShadowLimitB+0], s[sgprShadowLimitB+0], s74 // limit -= inc)
+s_waitcnt lgkmcnt(6)                               // wait for prior local read local write old=0, new=6 newLW=0 newLR=6
+v_mfma_f32_16x16x16_f16 acc[0:3], v[vgprValuB_X2_I0+0+2+0:vgprValuB_X2_I0+0+2+0+1], v[vgprValuA_X2_I0+0+2+0:vgprValuA_X2_I0+0+2+0+1], acc[0:3] // left value = acc[0+0:3+0]
+/*  mfmaIndex:7  */
+/* localReadsVacancy: latencyLeft 2 */
+s_subb_u32 s[sgprShadowLimitB+1], s[sgprShadowLimitB+1], s75 // limit -= inc)
+v_mfma_f32_16x16x16_f16 acc[4:7], v[vgprValuB_X2_I0+0+2+0:vgprValuB_X2_I0+0+2+0+1], v[vgprValuA_X2_I0+4+2+0:vgprValuA_X2_I0+4+2+0+1], acc[4:7] // left value = acc[4+0:7+0]
+/* numPrefetchIter=0 */
+/* dataAtIterA=-1 numReadsIterA=2 skipReadsIterA=2 readsPerIterA=2 */
+/* dataAtIterB=-1 numReadsIterB=2 skipReadsIterB=2 readsPerIterB=1 */
+
+/* iter 4 (swap and reset local write pointers iteration)  */
+/*  grEndMfmaIndex:9, lwStartMfmaIndex:9, lwEndMfmaIndex:9  */
+/*  numMfmaForLR:1, syncPlrMfmaIndex:12  */
+/*  mfmaIndex:8  */
+/* schedule remaining localreads for 1LDSB */
+/* localReadsVacancy: latencyLeft 2 */
+s_cmp_eq_u32 s[sgprShadowLimitB+1], 0              // are we within 2^32?
+/* 1 LDS buffer: read-sync-write */
+s_waitcnt lgkmcnt(0)
+s_barrier
+s_waitcnt lgkmcnt(3)                               // wait for prior local read local write old=0, new=3 newLW=0 newLR=3
```

Comment:
Redundant.
I will not point out every single redundant wait.... More redundant wait below...